client_plugin/drivers/shared/shared_driver.go

```diff
@@ -402,8 +401,171 @@ func (d *VolumeDriver) Path(r volume.Request) volume.Response {
+		return volume.Response{Mountpoint: d.GetMountPoint(r.Name)}
+	}
+
+	mountpoint, err := d.MountVolume(r.Name, "", "", false, true)
+	if err != nil {
+		log.WithFields(
+			log.Fields{"name": r.Name,
+				"error": err},
+		).Error("Failed to mount ")
+
+		refcnt, _ := d.DecrRefCount(r.Name)
+		if refcnt == 0 {
+			log.Infof("Detaching %s - it is not used anymore", r.Name)
+			// TODO: umount here
+		}
+		return volume.Response{Err: err.Error()}
+	}
+
+	return volume.Response{Mountpoint: mountpoint}
+}
+
+// MountVolume - Request attach and then mounts the volume.
+func (d *VolumeDriver) MountVolume(name string, fstype string, id string, isReadOnly bool, skipAttach bool) (string, error) {
+	mountpoint := d.GetMountPoint(name)
+	// First, make sure  that mountpoint exists.
+	err := fs.Mkdir(mountpoint)
+	if err != nil {
+		log.WithFields(
+			log.Fields{"name": name,
+				"dir": mountpoint},
+		).Error("Failed to make directory for volume mount ")
+		return mountpoint, err
+	}
+
+	// Increase GRef
+	log.Infof("Before AtomicIncr")
+	err = d.kvStore.AtomicIncr(kvstore.VolPrefixGRef + name)
+	if err != nil {
+		log.WithFields(
+			log.Fields{"name": name,
+				"error": err},
+		).Error("Failed to increase global refcount when processMount ")
+		return "", err
+	}
+
+	// Blocking wait till Mounted
+	info, err := d.kvStore.BlockingWaitAndGet(kvstore.VolPrefixState+name,
+		string(kvstore.VolStateMounted), kvstore.VolPrefixInfo+name)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to blocking wait for Mounted state. Error: %v.", err)
+		err = d.kvStore.AtomicDecr(kvstore.VolPrefixGRef + name)
+		if err != nil {
+			msg += fmt.Sprintf(" Also failed to decrease global refcount. Error: %v.", err)
+		}
+		log.WithFields(
+			log.Fields{"name": name,
+				"error": msg}).Error("")
+		return "", errors.New(msg)
+	}
+
+	// Start mounting
+	log.Infof("Volume state mounted, prepare to mounting locally")
+	var volRecord VolumeMetadata
+	// Unmarshal Info key
+	err = json.Unmarshal([]byte(info), &volRecord)
+	if err != nil {
+		log.WithFields(
+			log.Fields{"name": name,
+				"error": err},
+		).Error("Failed to unmarshal info data ")
+		return "", err
+	}
+
+	log.WithFields(
+		log.Fields{"name": name,
+			"Port":        volRecord.Port,
+			"ServiceName": volRecord.ServiceName,
+		}).Info("Get info for mounting ")
+	err = d.mountSharedVolume(name, mountpoint, &volRecord)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to mount shared volume. Error: %v.", err)
+		err = d.kvStore.AtomicDecr(kvstore.VolPrefixGRef + name)
+		if err != nil {
+			msg += fmt.Sprintf(" Also failed to decrease global refcount. Error: %v.", err)
+		}
+		log.WithFields(
+			log.Fields{"name": name,
+				"error": msg}).Error("")
+		return "", errors.New(msg)
+	}
+
+	return mountpoint, nil
+}
+
+// mountSharedVolume - mount the shared volume.
+func (d *VolumeDriver) mountSharedVolume(volName string, mountpoint string, volRecord *VolumeMetadata) error {
```

Comment:
Could add more comments mentioning exactly from where the shared volume is going to be mounted, prerequisite is that the volume must be in _mounted_ state.